### PR TITLE
Add tests for merging reports

### PR DESCRIPTION
**Pull request by chao2zhang**
This PR adds tests for #4192. The problem seems to be happening for Detekt/Android, but not Detekt/Jvm.

### CODE REVIEW

**Comment by cortinico** [comment]
> I would vote to move the tests back to Java 16 then. Until gradle supports Java 17.

I think this can be left on `blocked` till Gradle 7.3 is out which should be really soon (they're on RC)

**Comment by chao2zhang** [comment]
This PR is now unblocked since Gradle 7.3 supports Java 17 officially

**Comment by 3flex** [comment]
You might need to check and force push again, the diff is now empty...

**Review by BraisGabin — APPROVED** [review]

**Review by cortinico — APPROVED** [review]

File: detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/testkit/DslGradleRunner.kt [hunk 1]
```diff
@@ -124,6 +124,7 @@ class DslGradleRunner @Suppress("LongParameterList") constructor(
             add("--info")
             add("--build-cache")
             add("-Dorg.gradle.jvmargs=-Xmx2g -XX:MaxMetaspaceSize=1g")
+            add("--add-opens=java.base/java.io=ALL-UNNAMED")
```
Comment (3flex) [inline detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/testkit/DslGradleRunner.kt:127]: Why is this needed?

Comment (chao2zhang) [reply detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/testkit/DslGradleRunner.kt:127]: This is now back to a draft.  I will try to address the Java 17 failure
